packages/lesswrong/components/posts/PostsPage/AudioToggle.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useMemo } from 'react';
+import { registerComponent, Components } from '../../../lib/vulcan-lib';
+import { useCookiesWithConsent } from '@/components/hooks/useCookiesWithConsent';
+import moment from 'moment';
+import { PODCAST_TOOLTIP_SEEN_COOKIE } from '@/lib/cookies/cookies';
+import classNames from 'classnames';
+import { isLWorAF } from '@/lib/instanceSettings';
+import { isFriendlyUI } from '@/themes/forumTheme';
+
+const PODCAST_ICON_SIZE = isFriendlyUI ? 22 : 24;
+// some padding around the icon to make it look like a stateful toggle button
+const PODCAST_ICON_PADDING = isFriendlyUI ? 4 : 2
+
+const styles = (theme: ThemeType) => ({
+  togglePodcastContainer: {
+    alignSelf: 'center',
+    color: theme.palette.text.dim3,
+    height: PODCAST_ICON_SIZE,
+  },
+  audioIcon: {
+    width: PODCAST_ICON_SIZE + (PODCAST_ICON_PADDING * 2),
+    height: PODCAST_ICON_SIZE + (PODCAST_ICON_PADDING * 2),
+    transform: `translateY(-${PODCAST_ICON_PADDING}px)`,
+    padding: PODCAST_ICON_PADDING
+  },
+  audioNewFeaturePulse: {
+    top: PODCAST_ICON_PADDING * 1.5
+  },
+  audioIconOn: {
+    background: theme.palette.icon.dim05,
+    borderRadius: theme.borderRadius.small
+  },
+  audioIconDisabled: {
+    opacity: 0.5,
+    cursor: 'not-allowed'
+  }
+});
+
+const AudioToggle = ({classes, post, toggleEmbeddedPlayer, showEmbeddedPlayer}: {
+  classes: ClassesType<typeof styles>,
+  post: PostsWithNavigation|PostsWithNavigationAndRevision|PostsListWithVotes,
+  toggleEmbeddedPlayer?: () => void,
+  showEmbeddedPlayer?: boolean,
+}) => {
+  const { LWTooltip, NewFeaturePulse, ForumIcon } = Components;
+
+  const [cookies, setCookie] = useCookiesWithConsent([PODCAST_TOOLTIP_SEEN_COOKIE]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const cachedTooltipSeen = useMemo(() => cookies[PODCAST_TOOLTIP_SEEN_COOKIE], []);
+
+  useEffect(() => {
+    if(!cachedTooltipSeen) {
+      setCookie(PODCAST_TOOLTIP_SEEN_COOKIE, true, {
+        expires: moment().add(2, 'years').toDate(),
+      });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+  const audioIcon = <LWTooltip title={'Listen to this post'} className={classes.togglePodcastContainer}>
+    <a href="#" onClick={toggleEmbeddedPlayer}>
+      <ForumIcon icon="VolumeUp" className={classNames(classes.audioIcon, {[classes.audioIconOn]: showEmbeddedPlayer, [classes.audioIconDisabled]: !toggleEmbeddedPlayer})} />
+    </a>
+  </LWTooltip>
+
+  if (cachedTooltipSeen || isLWorAF) {
+    return audioIcon
```

Comment:
Thanks for pulling this out to a separate component! :)

I noticed that this shows the audio icon for events now, which you probably don't want on LW. Given that, and after playing around with it, I think I prefer for us to keep the previous behavior. Could you change it back, probably for all friendly UI sites?

<img width="330" alt="Screen Shot 2024-08-13 at 3 08 05 PM" src="https://github.com/user-attachments/assets/b6071b64-9a67-4ecb-b797-ea308c6958b1">

Reply:
Oh, hrm. That's an unintended side effect of my change. 

I think the actually correct way to deal with is to hide the icon in the places where it won't ever have the ability to play audio, and use the disabled-behavior in places where the icon briefly doesn't work but will work in ~2 seconds.

(We'd want to fix that on our end, at least, and if we fix that I'm guessing the resulting UI would be an improvement. I could still forumGate it if you want)

Reply:
Hmm ok if you are able to hide it in all the right places, that seems fine to leave ungated. A couple things I forgot to mention here:
1. Could you make it so that, if the button is disabled, the dark background never appears?
2. The icon is slightly misaligned on EAF - looks like a quick fix is to just set `height: 16` for `togglePodcastContainer` (that's what I did in the screenshot below)

<img width="668" alt="Screen Shot 2024-08-13 at 6 49 17 PM" src="https://github.com/user-attachments/assets/7a2e9dfc-449f-47d2-9e2f-64b128cd4595">